README.md

```diff
@@ -104,6 +104,42 @@ Here is a complete example:
 </body>
 ```
 
+<br/>
+However, in larger applications, to ensure the iframe and the build method are called in the same JavaScript event loop, you can create the iframe and set its src, then call build and capture its promise, before finally appending the iframe and awaiting the captured promise.
+
+```html
+<body>
+  <script type="module">
+    import {PublisherInterface} from 'https://unpkg.com/@chili-publish/publisher-interface@latest/dist/PublisherInterface.min.js';
+
+    document.addEventListener("DOMContentLoaded", async () => {
+      (async () => {
+        const iframe = document.createElement("iframe");
+        iframe.src = "https://example.chili-publish.online/example/editor_html.aspx?doc=3d178228-a9b9-49d0-90d9-c1c8f8b67f05&apiKey=Sczs1ruhiZcaFiqg0G07gMFMq07X+SG2o8KlW8oAeZGqoB1a0YkbeZU1wJK15aIhANgZmhg+13NQlxpBEq7Q==";
+        const publisherPromise = PublisherInterface.build(iframe);
+        document.body.appendChild(iframe);
+        const publisher = await publisherPromise;
+        const documentName = await publisher.getObject("document.name");
+        console.log(documentName);
+      })();
+    });
+  </script>
+</body>
+```
+## Debugging
+To enable debugging for the underlying library, pass {penpalDebug: true} as the buildOptions in your build function.
+
+```javascript
+PublisherInterface.build(iframe, {penpalDebug: true});
+```
+
+This will display messages in the console about Penpal and the communication between the iframe and the main page.
```

Comment:
Maybe "This will display messages in the console _from_ Penpal regarding the communication between the your page and the iframe.